learn/os_setup/choose_compiler.md

```diff
@@ -0,0 +1,18 @@
+---
+layout: book
+title: Choosing a compiler
+permalink: /learn/os_setup/choose_compiler
+---
+
+A comprehensive list of available compilers is provided in this website under [Compilers]({{site.baseurl}}/compilers). In this guide, we will focus only on those that fulfill the following criteria:
+1. Free to install and use
+2. Actively maintained/updated
+3. Production-level ready
+
+Out of those enlisted at the link above, and as of the time of writing this tutorial only two fully match our criteria, GFortran and OpenCoarrays, both members of the GNU Compiler Collection (GCC), while NVFortran and IFort do match under certain conditions. 
+
+NVFortran is part of the [NVIDIA HPC SDK formerly known as PGI Compilers & Tools](https://www.pgroup.com/index.htm). It comes with both free and paid versions. The current version 20.7 is available only for Linux x86_64 systems, while the Windows version *will be available at a later date.* It worths mentioning that nvfortran comes with NVIDIA CUDA Fortran, *a small set of extensions to Fortran that supports and is built upon the CUDA computing architecture*.
+
+IFort (Intel's Fortran Compiler) is not free in general. However, if you are a student, educator, or open-source contributor, you may [qualify for free software](https://software.intel.com/content/www/us/en/develop/articles/qualify-for-free-software.html).
+
+Both NVFortran(NVidia) and IFort(Intel) are very well documented and readers can find detailed information about installing and using them on their websites respectively. For that reason, in the next chapter, we will focus only on the open-source GCC compilers. In the future, when more compilers (e.g., the expected [Flang](https://github.com/llvm/llvm-project/tree/master/flang) and [LFortran](https://lfortran.org/)) fully meet our criteria, this guide will be updated accordingly.
```

Comment:
I was misunderstood. I did not imply that gfortran is production-level ready while ifort or nag are not. However, as a beginner tutorial, we cannot really suggest a commercial (non-free) compiler. That is why I still mention ifort, since, even if not free, in general, there is a high chance if you hold a valid academic e-mail to be qualified for free software. Then, of course, we cannot just have only the "free" criteria, I mean LFortran, is open-source and free, and under heavy development, but quoting your words it is still at pre-alpha stage. I hope you understand my line of thinking. I am trying to explain my three criteria.

Anyway, since "production-level ready" is probably not a good wording I will remove it. 

Speaking of compilers, I have a couple of questions about [Flang](https://github.com/flang-compiler/flang) a compiler that I have never used, neither am I aware of any projects built with it. Looking at it more carefully, it might deserve more attention and perhaps a separate page. For example, since it is mentioned at the official AMD [site](https://developer.amd.com/amd-aocc/), I guess it is a mature compiler? I am just asking because initially, I thought it was like LFortran, at an initial stage of development. I would appreciate any information on Flang so I can incorporate it into the tutorial.

Reply:
In my opinion, the best way forward is to leave it to the compiler vendors to advertise for their compilers and we simply present the information.

For the tutorial, I suggest we use GFortran. It is the most mature of the open source compilers currently. We can say something like:

> We use GFortran for the tutorials as it is a mature open source compiler. Users are encouraged to also try other open source and commercial compilers from the compiler list.

Or something like that.

LFortran is not production ready, but we are getting there.

LLVM Flang is also under heavy development. You can find the latest updates for both Flang and LFortran in the Fortran newsletter, the latest is:

https://fortran-lang.org/newsletter/2020/10/01/Fortran-Newsletter-October-2020/

The AMD version of Flang might be based on the Classic Flang, although I could be wrong. The Classic Flang is based on a mature production PGI compiler. Although it can't compile our production codes, so I personally do not use it, but for the tutorial it should work.

Anyway, it's really hard to write objective review of the compilers, so the best we did so far is to simply let the compiler vendors themselves advertise what they want at:

https://fortran-lang.org/compilers/

and I think that works well.

Btw., what you wrote about LFortran is accurate, I am not offended at all. But I feel some other vendors might not like that we either didn't mention them, or passed judgement on them. We want everybody to feel included at fortran-lang.org, as a neutral platform.

Reply:
Just a typo: "worths" should be "is worth"

Reply:
@certik I did some changes please check again.